docs/design/eth-subscribe.md

```diff
@@ -0,0 +1,366 @@
+
+### Subscription Types
+
+|   Subscription Type    |                                  Description                                   |                           Support                           |
+| :--------------------: | :----------------------------------------------------------------------------: |:-----------------------------------------------------------:|
+|          logs          |      Emits logs attached to a new block that match certain topic filters.      |                       Yes, initially                        |
+|        newHeads        |               Emits new blocks that are added to the blockchain.               |                 No, maybe in future release                 |
+| newPendingTransactions | Emits transaction hashes that are sent to the network and marked as `pending`. | No, as Hedera does not have pending transactions on a node. |
+
+### Initiating a subscription
+
+#### Request
+
+```javascript
+// initiate websocket stream first
+wscat -c wss://<env>.hashio.io/api
+
+// then call subscription
+{"jsonrpc":"2.0","id": 1, "method": "eth_subscribe", "params": [SUBSCRIPTION_TYPE, PARAMS]}
+```
+
+#### Response
+
+```javascript
+{"jsonrpc":"2.0","id": 1, "result": SUBSCRIPTION_ID}
+```
+
+### Logs
+
+     The `logs` subscription type emits logs that match a specified topic filter and are included in newly added blocks.
+
+#### Parameters
+
+An object with the following fields:
+
+- address (optional): [`string`] or [`array of strings`] Singular address or array of addresses. Only logs created from one of these addresses will be emitted.
+- topics: an array of topic specifiers.
+  - Each topic specifier is either null, a single string, or an array of strings.
+  - For every non-null topic, a log will be emitted when activity associated with that topic occurs.
+
+#### Request
+
+```javascript
+// initiate websocket stream first
+wscat -c wss://<env>.hashio.io/api
+
+// then call subscription
+{"jsonrpc":"2.0","id": 1, "method": "eth_subscribe", "params": ["logs", {"address": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48", "topics": ["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]}]}
+```
+
+#### Response
+
+This response is sent whenever new data is available.
+
+```javascript
+{
+  "method": "eth_subscription",
+  "params": {
+    "subscription": SUBSCRIPTION_ID,
+    "result": {
+      "address": "0x8320fe7702b96808f7bbc0d4a888ed1468216cfd",
+      "blockHash": "0x61cdb2a09ab99abf791d474f20c2ea89bf8de2923a2d42bb49944c8c993cbf04",
+      "blockNumber": "0x29e87",
+      "data": "0x00000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000003",
+      "logIndex": "0x0",
+      "topics": ["0xd78a0cb8bb633d06981248b816e7bd33c2a35a6089241d099fa519e361cab902"],
+      "transactionHash": "0xe044554a0a55067caafd07f8020ab9f2af60bdfe337e395ecd84b4877a3d1ab4",
+      "transactionIndex": "0x0"
+    }
+  }
+}
+```
+
+### newHeads
```

Comment:
Drop the `newHeads` and `newPendingTransactions` sections since they aren't part of the design